gcloud-java-storage/src/main/java/com/google/gcloud/storage/Bucket.java

```diff
@@ -172,17 +175,41 @@ public Blob get(String blob, BlobSourceOption... options) {
   }
 
   /**
-   * Creates a new blob in this bucket.
+   * Creates a new blob in this bucket. Direct upload is used to upload {@code content}.
+   * For large content, {@link Blob#writer(com.google.gcloud.storage.Storage.BlobWriteOption...)}
+   * is recommended as it uses resumable upload. MD5 and CRC32C hashes of {@code content} are
+   * computed and used for validating transferred data.
    * 
    * @param blob a blob name
    * @param content the blob content
+   * @param contentType the blob content type
```

Comment:
Lets document that this can be null (and if so reference the default value that will be picked).